File: PDOO/Napakalaki/src/NapakalakiGame/BadConsequence.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

import java.util.ArrayList;

/**
 *
 * @author Alejandro
 */
public abstract class BadConsequence {

    static final int MAXTREASURES = 10;
    private String text;
    private int levels;
    private int nVisibleTreasures = 0;
    private int nHiddenTreasures = 0;
    private boolean death;
    private ArrayList<TreasureKind> specificHiddenTreasures = new ArrayList();
    private ArrayList<TreasureKind> specificVisibleTreasures = new ArrayList();

    public BadConsequence(String t, int l) {
        this.text = t;
        this.levels = l;
    }

    public String getText() {
        return text;
    }

    public int getLevels() {
        return levels;
    }

    public abstract boolean isEmpty();
    
    public abstract int getNVisibleTreasures();
    
    public abstract int getNHiddenTreasures();
    
    public abstract ArrayList<TreasureKind> getSpecificVisibleTreasures();
    
    public abstract ArrayList<TreasureKind> getSpecificHiddenTreasures();
    
    public abstract boolean getDeath();

    public abstract void substractVisibleTreasure(Treasure t);

    public abstract void substractHiddenTreasure(Treasure t);

    public abstract BadConsequence adjustToFitTreasureLists(ArrayList<Treasure> v, ArrayList<Treasure> h);

    public String toString() {
        return "Text = " + text + " Levels = " + Integer.toString(levels);
    }
}

File: PDOO/Napakalaki-Ruby/lib/bad_consequence.rb
#encoding: UTF-8

# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module NapakalakiGame

  require_relative "treasure_kind.rb"
  require_relative "player.rb"

  class BadConsequence
    
    @@MAXTREASURES=10
    def initialize (aText, someLevels)
      @text = aText
      @levels = someLevels
    end
  
    def getText
      return @text
    end
  
    def getLevels
      return @levels
    end
  
    def to_s
      return "Text: #{@text} Levels: #{@levels} "
    end

  end

end
